### DISCUSSION TITLE
Which authentication method is best practice?

### DISCUSSION THREAD

**Discussion by david-engelmann** [tpope/vim-fugitive#2073] [Q&A] [answered]
I'm looking to push to a remote branch. Whenever I try to push a commit, It prompts me to enter a Username (See Attached Screenshot).

![image](https://user-images.githubusercontent.com/81196676/199586986-87c3f487-8d5d-459b-baf0-e84674e100f0.png)
 
After entering my Username, it prompts me to input my password. Regular passwords have been depreciated.

![image](https://user-images.githubusercontent.com/81196676/199587239-310b2601-b405-42a3-bdb2-7a1ef0e167cb.png)

Is the best approach to set up a "Fine-grained personal access tokens" to use as a password? Would it be better to set it up with "SSH key passphrases". 

Optimally, I'm not prompted to input anything when I push commits. With that being said, Is SSH preferred?

**Comment by tpope** [comment, marked as answer]
I personally use SSH, with an agent to avoid needing to enter the passphrase every time. Tokens will work too.

**Reply by david-engelmann** [reply]
Added a ~/.ssh folder to my windows user and ubuntu user, permissioned it on GitHub and now we flying. Thank you for all that you do!
